### ISSUE TITLE
[Feature request] Bottom bar for book status

### ISSUE THREAD

**Issue by ngocanhtve** [tfonteyn/NeverTooManyBooks#44]
I'm wondering if this feature request conflicts with the design language or the original intended use of this app!
If so please let me know and please forgive me!

First there needs to be the bottom bar, where a few tabs are placed.

One tab for _finished_ books, one for _reading_ books, and one for _want to read_ books. And one for _Lending_ books - maybe.

I know I'm asking for the feature of a book tracking app, while this is a book collection app, but who knows, maybe the feature and UI will work well together.

Finally, this app has a very nice Material You design, it would be more perfect if it also supported Dynamic Accent Colors.

Thank for your great app!

**Comment by tfonteyn** [comment]
I've split the Dynamic Accent Colors request into #46 

I've seen the type of interface/bottom-bar you describe elsewhere.
But it does indeed not really fit the concept of the app (but read on until the end!)

The (current) concept is based on Books on Bookshelves  (which explains my use of 'BoB' in source code comments); and a Bookshelf has a Style.

A Bookshelf/Style combination can basically be anything you like (by configuring the Style).
The Bookshelf name can also be seen as what you want to do with the books on that shelf.

Unlike real life, Books can sit on multiple Bookshelves at the same time.
Suppose you started simply by adding all the books you have to the "My Books" default shelf.

You can now for example create a shelf "To Read" and assign books to it - those books would then 'sit' on both the "My Books" as you own them + on the "To Read" as you still need to read them.

Likewise, create a Bookshelf "I have read" and.... same thing as above.

As an alternative, don't forget you can set filters on a Bookshelf, to temporarily or permanently only see Books matching the criteria you set. As an example, there is a filter "Read" which can be set to "Reading, Unread, Read"; or "Lending" with the options "Lend out" and "Available". There are 10 possible filters for now. If you have an idea for a new one, then just log an RFE and I'll look into it.

Creating a specific menu (bottom bar) as described is basically: create buttons which serve as short-cuts to a limited set of Bookshelves.

In other words:

    One tab for finished books, one for reading books, and one for want to read books.
    And one for Lending books - maybe.

Basically, create 4 Bookshelves, assign the books as needed and/or use filters, and you get the same functionality except that you need to use the drop-down to select the desired shelf instead of a button.

_But... and now I'm going to contradict myself!_

Suppose I create a bottom-bar (swipe-up to view) with configurable buttons. Each button would apply a set of filters to the **current** Bookshelf. In other words, they would become short-cuts to filters.

That would, I think, stay true to the BoB design, but at the same time allow on-the-fly applying of filters such as "finished books, one for reading books, and one  ..." as you describe.

I hope it makes sense; but all comments are welcome.

TLDR: ok, I'll look into implementing this but I cannot make a hard promise until I've experimented a bit.

**Comment by tfonteyn** [comment]
I need some extra input on

> One tab for finished books, one for reading books, and one for want to read books. And one for Lending books - maybe.

- Finished ("Read")
- Want-to-read ("Unread")
- Lending ("Lend out")

are pretty clear and useful.

But "one for reading books" makes me wonder: I generally have 3 or 4 books on the go (one with short-stories, one novel, and 1 or 2 non-fiction) but not more. I wonder how many people will actually *need* a "list" of books that they are reading?

any thoughts?

I'm still speculating of simply providing 3 to 4 freely configurable buttons for filters and leave it to users what they consider useful, but I'm thinking of some defaults as you defined them.

**Comment by ngocanhtve** [comment]
> any thoughts?

You're right. I didn't think about dedicating a tab for 1 or 2 reading books, what a waste of space.

> I'm still speculating of simply providing 3 to 4 freely configurable buttons for filters and leave it to users what they consider useful, but I'm thinking of some defaults as you defined them.

I think this will be a good development direction. So I would suggest a tab for bookshelves, a tab for statistics. The remaining two tabs can be reserved for the settings menu or the user can choose to put in a custom filter, for example.

**Comment by ngocanhtve** [comment]
I just thought of a few more. 

Move _this menu_:

![](https://i.imgur.com/gWf7qQ9.png)

Go to the _more_ tab like the example in the image below:

![](https://i.imgur.com/GhAr2Bc.png)

**Comment by tfonteyn** [comment]
moved to the backlog for now.
